test/agents/test_conversational_agent.py

```diff
@@ -54,3 +56,78 @@ def test_run():
         agent.run = MagicMock(return_value="Hello")
         assert agent.run("query") == "Hello"
         agent.run.assert_called_once_with("query")
+
+
+@pytest.fixture
+def prompt_node():
+    prompt_node = PromptNode("google/flan-t5-xxl", api_key="fake_key", max_length=256)
+    return prompt_node
+
+
+@pytest.mark.unit
+def test_init(prompt_node):
+    agent = ConversationalAgent(prompt_node)
+
+    # Test normal case
+    assert isinstance(agent.memory, ConversationMemory)
+    assert callable(agent.prompt_parameters_resolver)
+    assert agent.max_steps == 5
+    assert agent.final_answer_pattern == r"Final Answer\s*:\s*(.*)"
+    assert agent.prompt_template.name == "conversational-agent"
+
+
+@pytest.mark.unit
+def test_tooling(prompt_node):
+    agent = ConversationalAgent(prompt_node)
+    # ConversationalAgentWithTools has no tools by default
+    assert not agent.tm.tools
+
+    # but can add tools
+    agent.tm.add_tool(Tool("ExampleTool", lambda x: x, description="Example tool"))
+    assert agent.tm.tools
+    assert agent.tm.tools["ExampleTool"].name == "ExampleTool"
+
+
+@pytest.mark.unit
+def test_agent_with_memory(prompt_node):
+    # Test with summary memory
+    agent = ConversationalAgent(prompt_node, memory=ConversationSummaryMemory(prompt_node))
+    assert isinstance(agent.memory, ConversationSummaryMemory)
```

Comment:
For completeness, we could also check what memory the `agent = ConversationalAgent(prompt_node)` gets assigned automatically. Should be `ConversationMemory`.